examples/animation/main.rs

```diff
@@ -63,30 +85,145 @@ impl State for Example {
+                            } else {
+                                control_set.add_animation(
+                                    self.current_animation,
+                                    &animation,
+                                    EndControl::Normal,
+                                    self.rate,
+                                    AnimationCommand::Start,
+                                );
+                            }
+                        }
+
+                        Some(VirtualKeyCode::D) => {
+                            {
+                                let animation = world
+                                    .read::<AnimationSet<AnimationId, Transform>>()
+                                    .get(self.sphere.unwrap())
+                                    .and_then(|s| s.get(&AnimationId::Translate))
+                                    .cloned()
+                                    .unwrap();
+                                let mut sets = world.write();
+                                let mut control_set = get_animation_set::<AnimationId, Transform>(
+                                    &mut sets,
+                                    self.sphere.unwrap().clone(),
+                                );
+                                control_set.add_animation(
+                                    AnimationId::Translate,
+                                    &animation,
+                                    EndControl::Normal,
+                                    self.rate,
+                                    AnimationCommand::Start,
+                                );
+                            }
+                            {
+                                let animation = world
+                                    .read::<AnimationSet<AnimationId, Transform>>()
+                                    .get(self.sphere.unwrap())
+                                    .and_then(|s| s.get(&AnimationId::Rotate))
+                                    .cloned()
+                                    .unwrap();
+                                let mut sets = world.write();
+                                let mut control_set = get_animation_set::<AnimationId, Transform>(
+                                    &mut sets,
+                                    self.sphere.unwrap().clone(),
+                                );
+                                control_set.add_deferred_animation(
+                                    AnimationId::Rotate,
+                                    &animation,
+                                    EndControl::Normal,
+                                    self.rate,
+                                    AnimationCommand::Start,
+                                    AnimationId::Translate,
+                                    DeferStartRelation::End,
+                                );
+                            }
+                            {
+                                let animation = world
```

Comment:
Any way to reuse this? The code duplication is getting out of control.

Reply:
Done.